docs/openvpn.md

```diff
@@ -0,0 +1,164 @@
+- create a privileged container in Proxmox and enable nesting in it;
+- deploy an OpenVPN image for Docker within that container;
+- enable IP forwarding in the hypervisor and all the nested containers;
+- add some settings to the network configuration OpenVPN server sends to its clients:
+  - route to the local network via the OpenVPN server (necessary),
+  - local DNS server (optional).
+
+## Step 1. Enable IP forwarding on Proxmox
+
+```shell
+tee /etc/sysctl.d/ip_forward.conf <<EOF
+net.ipv4.ip_forward=1
+EOF
+# reload rules from /etc/sysctl.d
+sysctl --system
+```
+
+## Step 2. Set up a Debian container
+
+On this stage, it's assumed you already have a Debian container image in Proxmox.
+
+### Create the container
+
+- Start creating a new container and disable the `Unprivileged container` option on the `General` tab.
+- Enable DHCP on the `Network` tab.
+- The container needs at least 1 vCPU, 128 MB of RAM, 0 MB of swap, and 5 GB of disk space.
+
+Don't start the container after creating. Docker needs to have nesting enabled, so:
+- open the file `/etc/pve/lxc/<CONTAINER_ID>.conf` on the Proxmox machine,
+- enable nesting: `features: nesting=1`,
+- enable the capability to create devices:
+  ```
+  lxc.mount.entry: /dev/net/tun dev/net/tun none bind,create=file
+  lxc.cgroup2.devices.allow: c 10:200 rwm 
+  ```
+  
+Your container config file will finally look like this:
+```
+arch: amd64
+cores: 1
+features: nesting=1
+hostname: openvpn
+memory: 256
+net0: name=eth0,bridge=vmbr1,firewall=1,hwaddr=FF:…:99,ip=dhcp,type=veth
+ostype: debian
+rootfs: …,size=5G
+swap: 0
+lxc.mount.entry: /dev/net/tun dev/net/tun none bind,create=file
+lxc.cgroup2.devices.allow: c 10:200 rwm
+```
+
+Start the container.
+
+### Enable IP forwarding
+
+```shell
+tee /etc/sysctl.d/ip_forward.conf <<EOF
+net.ipv4.ip_forward=1
+EOF
+# reload rules from /etc/sysctl.d
+sysctl --system
+```
+
+## Step 3. Deploy OpenVPN
+
+All the following commands are run in the Debian container shell.
+
+### Install Docker on the Debian container
+
+```shell
+apt update && apt install curl -y && curl https://get.docker.com | bash
+```
+
+### Configure the server image
+
+#### Create the files for container settings
+
+```shell
+tee sysctl.conf <<EOF
+net.ipv4.ip_forward=1
+EOF
+
+tee docker-compose.yml <<EOF
+services:
+  openvpn:
+    cap_add:
+     - NET_ADMIN
+    image: kylemanna/openvpn
+    container_name: openvpn
+    ports:
+     - 1194:1194/udp
+    restart: always
+    volumes:
+     - ./openvpn-data/conf:/etc/openvpn
+     - ./sysctl.conf:/etc/sysctl.conf
+EOF
+```
+
+#### Generate and edit the server settings
+
+```shell
+export OPENVPN_SERVER="your_server_public_address"
+docker compose run --rm openvpn ovpn_genconfig -u udp://$OPENVPN_SERVER
+```
+
+You will have to manually edit `openvpn-data/conf/openvpn.conf` because OpenVPN doesn't support config directories:
+
+```shell
+nano openvpn-data/conf/openvpn.conf
```

Comment:
I see that your setup generates a server config file that refers to certificates in `/etc/openvpn/pki/` but `ovpn_genconfig` doesn't seem to generate them.
Are they generated on the first start?